solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttp2SolrClient.java

```diff
@@ -258,4 +299,32 @@ public void onFailure(Throwable oe) {
               }
             });
   }
+
+  public static class Builder {
+    private final Http2SolrClient http2Client;
+    private final String[] baseSolrUrls;
+    private ResponseParser responseParser;
+    private RequestWriter requestWriter;
+
+    public Builder(Http2SolrClient http2Client, String... baseSolrUrls) {
+      this.http2Client = http2Client;
+      this.baseSolrUrls = baseSolrUrls;
+    }
+
+    /** Provides a {@link ResponseParser} for created clients to use when handling requests. */
+    public LBHttp2SolrClient.Builder withResponseParser(ResponseParser responseParser) {
+      this.responseParser = responseParser;
+      return this;
+    }
+
+    /** Provides a {@link RequestWriter} for created clients to use when handing requests. */
+    public LBHttp2SolrClient.Builder withRequestWriter(RequestWriter requestWriter) {
+      this.requestWriter = requestWriter;
+      return this;
+    }
```

Comment:
are you suggesting that maybe these methods aren't actually needed?  because you would never set them in the LBHttp2SolrClient, you would do this in the Http2SolrClient that underlies it?   That does intuitively make sense...

Reply:
there is a lot of overlap with the other discussion thread (I created some of that confusion, sorry). I agree with @dsmiley here. if we don't expect the response parser to be different (which it doesn't look like we do), we can assume the client is already 'well configured' via its own builder and do not expose these extra methods on this builder.